backend-tests/tests/test_deployments.py

```diff
@@ -0,0 +1,510 @@
+               ("artifact", (filename, open(filename, 'rb'),
+                             "application/octet-stream"))))
+    assert r.status_code == 201
+
+def create_tenant_test_setup(user_name, tenant_name, nr_deployments=3, nr_devices=100):
+    """
+    Creates a tenant, and a user belonging to the tenant belonging to the user
+    with 'nr_deployments', and 'nr_devices'
+    """
+    tenant = create_tenant(tenant_name)
+    user = create_tenant_user(user_name, tenant)
+    r = ApiClient(useradm.URL_MGMT).call(
+        'POST', useradm.URL_LOGIN, auth=(user.name, user.pwd))
+    assert r.status_code == 200
+    user.utoken = r.text
+    tenant.users = [user]
+    upload_image("/tests/test-artifact.mender", user.utoken)
+    # Create three deployments for the user
+    for i in range(nr_deployments):
+        request_body = {
+            "name": str(i) + "st-dummy-deployment",
+            "artifact_name": "deployments-phase-testing",
+            "devices": ["uuid not needed" + str(i) for i in range(nr_devices)],
+        }
+        resp = deploymentclient.with_auth(user.utoken).call(
+            'POST',
+            '/deployments',
+            body=request_body,
+        )
+        assert resp.status_code == 201
+    # Verify that the 'nr_deployments' expected deployments have been created
+    resp = deploymentclient.with_auth(user.utoken).call('GET', '/deployments')
+    assert resp.status_code == 200
+    assert len(resp.json()) == nr_deployments
+    return tenant
+
+
+@pytest.fixture(scope='function')
+def setup_deployments_enterprise_test(clean_mongo,
+                                      existing_deployments=3,
+                                      nr_devices=100):
+    """
+    Creates two tenants, with one user each, where each user has three deployments,
+    and a hundred devices each.
+    """
+    tenant1 = create_tenant_test_setup('bugs-bunny', 'acme')
+    # Add a second tenant to make sure that the functionality does not interfere with other tenants
+    tenant2 = create_tenant_test_setup('road-runner', 'indiedev')
+    # Create 'existing_deployments' predefined deployments to act as noise for the server to handle
+    # for both users
+    return tenant1, tenant2
+
+
+class TestDeploymentsEndpointEnterprise(object):
+    test_cases = [
+        # One phase:
+        #     + start_time
+        #     + batch_size
+        (
+            {
+                "name": "One phase, with start time, and full batch size",
+                "artifact_name": "deployments-phase-testing",
+                "devices": ["dummyuuid" + str(i) for i in range(100)],
+                "phases": [{
+                    "start_ts": str((datetime.utcnow() +
+                         timedelta(days=1)).isoformat("T")) + "Z",
+                    "batch_size": 100,
+                }],
+            },
+            {
+                "name": "One phase, with start time, and full batch size",
```

Comment:
with start time?

Reply:
Misleading how?

Reply:
the name suggests that there is one phase WITH START TIME and batch size equal to 100, but there is one phase with batch size equal to 100 and WITHOUT START TIME